### PR TITLE
Merging more with updates

### PR DESCRIPTION
**Pull request by jordancarlin**
Combine additional "with" updates per #583

### CODE REVIEW

File: model/riscv_sys_regs.sail [hunk 1]
```diff
@@ -271,11 +272,7 @@ function legalize_mstatus(o : Mstatus, v : xlenbits) -> Mstatus = {
   let m = if xlen == 64 then {
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:272]: You can drop the `m`, i.e. just end with

```
if xlen == 64 then {
...
} else m
```

File: model/riscv_sys_regs.sail [hunk 2]
```diff
@@ -246,22 +246,23 @@ function legalize_mstatus(o : Mstatus, v : xlenbits) -> Mstatus = {
...
    * but only if Zfinx isn't enabled.
    * FIXME: This should be made a platform parameter.
    */
-  let m = if sys_enable_zfinx() then [m with FS = extStatus_to_bits(Off)] else m;
   let dirty = extStatus_of_bits(m[FS]) == Dirty | extStatus_of_bits(m[XS]) == Dirty |
               extStatus_of_bits(m[VS]) == Dirty;
-  let m = [m with SD = bool_to_bits(dirty)];
+
+  /* Legalize MPP */
+  let m = [m with MPP = if have_privLevel(m[MPP]) then m[MPP]
```
Comment (Timmmm) [inline model/riscv_sys_regs.sail:258]: I dunno the exact best way to format these but elsewhere I've been doing it like this:

```
let m = [m with
  MPP = ...
  ...
  MPRV = ...,
];
```

Also note the trailing comma.

Hopefully one day we can use `sail -format` or whatever.

**Review by Timmmm — CHANGES_REQUESTED** [review]

**Review by Timmmm — APPROVED** [review]

**Comment by Timmmm** [comment]
LGTM; I'll merge in a couple of days if nobody objects.

**Review by arichardson — APPROVED** [review]

File: model/riscv_sys_regs.sail (hunk 2 above)
Comment (jordancarlin) [reply model/riscv_sys_regs.sail]: I believe I changed it to use that style already, but if you think further modifications would help let me know.

Comment (arichardson) [reply model/riscv_sys_regs.sail]: Sorry that was an old pending comment that was only submitted when I added my review. This has been addressed.

**Pull request merged**